module/os/linux/zfs/zfs_acl.c

```diff
@@ -2465,6 +2465,8 @@ zfs_zaccess_trivial(znode_t *zp, uint32_t *working_mode, cred_t *cr,
 		return (unmapped ? SET_ERROR(EPERM) : 0);
 	}
 
+	rw_enter(&zp->z_gperm_lock, RW_READER);
```

Comment:
This lock is only ever taken as a reader and doesn't provide any protection.  Should this have been a RW_WRITER?  Was the intent here to serial calls to `generic_permission` or was it to safely set/clear `zp->z_lockedxattr `.  If the latter, couldn't we instead us the `zp->z_lock`.

Reply:
Agreed. @amotin also pointed it out and I am working on an alternative solution. The lock was supposed to protect the ```generic_permission()``` API as ```zp->z_lockedxattr``` is global to each znode. So, the write locks on every permission check do not seem to be a good idea.